File: services/circleguard-identity-service/src/test/java/com/circleguard/identity/util/IdentityEncryptionConverterTest.java
package com.circleguard.identity.util;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class IdentityEncryptionConverterTest {

    private IdentityEncryptionConverter converter;
    private final String secret = "test-secret-32-chars-long-123456";
    private final String salt = "deadbeef";

    @BeforeEach
    void setUp() {
        converter = new IdentityEncryptionConverter(secret, salt);
    }

    @Test
    void shouldEncryptAndDecryptSuccessfully() {
        String original = "user@example.com";
        byte[] encrypted = converter.convertToDatabaseColumn(original);
        
        assertNotNull(encrypted);
        assertNotEquals(original, new String(encrypted));
        
        String decrypted = converter.convertToEntityAttribute(encrypted);
        assertEquals(original, decrypted);
    }

    @Test
    void shouldHandleNulls() {
        assertNull(converter.convertToDatabaseColumn(null));
        assertNull(converter.convertToEntityAttribute(null));
    }

    @Test
    void differentPlaintextInputsProduceDifferentCiphertext() {
        byte[] encrypted1 = converter.convertToDatabaseColumn("user1@university.edu");
        byte[] encrypted2 = converter.convertToDatabaseColumn("user2@university.edu");

        assertFalse(java.util.Arrays.equals(encrypted1, encrypted2));
    }
}

File: services/circleguard-identity-service/src/test/kotlin/com/circleguard/identity/util/IdentityEncryptionConverterTest.kt
package com.circleguard.identity.util

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class IdentityEncryptionConverterTest {

    // Use the same values as application.yml (vault.secret / vault.salt)
    private val converter = IdentityEncryptionConverter(
        "746573742d7365637265742d33322d63686172732d6c6f6e672d313233343536",
        "deadbeef"
    )

    @Test
    fun `encrypt then decrypt round-trip preserves original value`() {
        val original = "john.doe@university.edu"

        val encrypted = converter.convertToDatabaseColumn(original)
        val decrypted = converter.convertToEntityAttribute(encrypted)

        assertEquals(original, decrypted)
    }

    @Test
    fun `different plaintext inputs produce different ciphertext`() {
        val encrypted1 = converter.convertToDatabaseColumn("user1@university.edu")
        val encrypted2 = converter.convertToDatabaseColumn("user2@university.edu")

        assertFalse(encrypted1.contentEquals(encrypted2))
    }
}
